packages/massa-proto-cli/src/ts-gen.ts

```diff
@@ -1,23 +1,46 @@
 import { ProtoFile } from './protobuf';
-import { writeFileSync } from 'fs';
+import { writeFileSync, renameSync, existsSync } from 'fs';
 import { execSync } from 'child_process';
 import * as returnType from './protoTypes.json';
+import { resolve, relative, join, dirname } from 'path';
+
 
 /**
- * Compile protofile to TypeScript helper class.
+ * Compile proto file to TypeScript helper class.
  *
  * @remarks
- * - If the @see helperFilePath is the relative path, it should be based on the .proto file path
+ * - The ts helper file is generated in the folder 'proto_build' at the same location as your current terminal.
  * - If the @see protoFilePath is the relative path, it should be based on the location of your terminal
  * 
  * @param protoFilePath - The path to the proto file.
- * @param helperFilePath - The path to save the helper file (ts).
  */
 export function compileProtoToTSHelper(
   protoFilePath: string,
-  helperFilePath: string,
+  functionName: string,
 ): void {
-  execSync(`npx protoc --ts_out="${helperFilePath}" ${protoFilePath}`, { stdio: 'inherit' });
+  // if needed, rename the proto file to 'functionName.proto'
+  if (!protoFilePath.endsWith(functionName + '.proto')) {
+    const directory = dirname(protoFilePath);
+    const newProtoFilePath = join(directory, functionName + '.proto');
+    renameSync(protoFilePath, newProtoFilePath);
+    protoFilePath = newProtoFilePath;
+  }
```

Comment:
Why not throwing an exception instead? Why should we silently rename the file? Is the .proto extension missing an expected behavior?

Reply:
The goal here is to detect if the name of the file. proto has been modified and no longer corresponds to the name of the function that its content represents in order to generate a helper corresponding to the correct name of the SC function.

Afterwards, renaming the proto file is not a good idea. It would better to rename the helper that is generated rather than the proto file itself